### DISCUSSION TITLE
Slow Download?

### DISCUSSION THREAD

**Discussion by wes1993** [tgdrive/teldrive#183] [Q&A]
Hello,
I have configured 4 bots but when i try to download some file i have the download speed limited at 5MBs but my connection is 1000Mbps

If i try to download the same file using the Telegram WEB K version (https://web.telegram.org/k/) the problem is not present.

I have a premium account.

Best Regards
Stefano

**Comment by wes1993** [comment]
I'm not sure if this could help or give some hint:

https://www.reddit.com/r/Telegram/comments/vh7kgj/comment/jbcvr25/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by magnumrabello** [comment]
I used 1 bot on hosted on oracle vps on brazil, i have 50~80mb (see on nload), when i do ftp/samba over external ip, i have 15~20mb and over zerotier i have 45~50mb. when i try 2 bots, refused my connections, i don't have premium. I hope this information helps.

**Reply by aZuiZui** [reply]
If you are using bots, the user accounts is only used as a pointer to access the file. Bots are used to transfer files.

New bots are blocked from transferring files by the telegram server so you will have to wait I think documentation says 20 minutes, but I find 24hrs is safe.

You can add the bot but remove from teldrive while waiting for the waiting time to avoid the invalid bot issue. After 24 add the bot back into teldrive.

Concerning the slow speed, this is random for me too. Some bots are fast some are slow, note the slow bot and remove it from the pool. I think it's to do with the location of the bot.

I have other issues, it takes a long time to start the initial open of the file, example if I test the file on Dropbox it opens instantly on teldrive it takes up to 30s to start a stream.

**Comment by aZuiZui** [comment]
I read on another similar teldrive as a storage.

Speed Test – Speed testing added for all bots on each file to optimize streaming performance.

DC-Aware Bot Selection – The underlying streaming load-balancer now ensures that a bot located in the exact same Data Center as the media file is prioritized. This substantially minimizes cross-DC round-trips and timeouts.

I don't know how he is doing this but might be the answer to poor performance. Maybe the devs can help implement something similar?
